ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java

```diff
@@ -71,6 +76,33 @@ public static ReactFontManager getInstance() {
     return typeface;
   }
 
+  public @Nullable Typeface getTypeface(
+    String fontFamilyName,
+    int style,
+    Context context) {
+    Typeface font = mTypeCache.get(fontFamilyName);
+
+    if (font != null) {
+      return Typeface.create(
+        font,
+        style
+      );
+    }
+
+    int fontId = context.getResources().getIdentifier(fontFamilyName, "font", context.getPackageName());
```

Comment:
So `getIdentifier()` is problematic in a big app that has many resources. This could contribute to non-trivial msec, just trying to crack open the resources, then looking it up from the APK. Rather than calling this for all scenarios, we'd instead build a "provider" pattern where:
* each app decides if it wants to provide a custom fonts
* if no provider is given, avoid asking `getIdentifier()` completely
* if given, ask the provider for a resourceId (int), then use it

Now the implementation of the provider needs to be a switch statement for the best performance. Example:

```
int getCustomFontProvider(name) {
  switch(name) {
    case "mine":
       return R.font.mine;
    ...
```

Referencing `R.font.*` directly is fast and efficient, because it points exactly to an int resource Id. The only problem is, that each app needs to implement this code themselves, because each app has different set of custom fonts.

What do you think?

Reply:
cc @cpojer @mdvacca